build-tools/automation/build.groovy

```diff
@@ -10,12 +10,24 @@ def isPr = false                // Default to CI
 
 def buildTarget = 'jenkins'
 
-def stageWithTimeout(stageName, timeoutValue, timeoutUnit, directory, fatal, Closure body) {
+def stageWithTimeout(stageName, timeoutValue, timeoutUnit, directory, fatal, ctAttempts, Closure body) {
```

Comment:
I don't know much about groovy, but could you use a default parameter for `ctAttempts`: http://mrhaki.blogspot.com/2009/09/groovy-goodness-parameters-with-default.html

Then you wouldn't need to pass 0, like all these have:
```diff
-        stageWithTimeout('init', 30, 'SECONDS', XADir, true) {    // Typically takes less than a second
+        stageWithTimeout('init', 30, 'SECONDS', XADir, true, 0) {    // Typically takes less than a second
```

And maybe `retries` is a better name?

```suggestion
def stageWithTimeout(stageName, timeoutValue, timeoutUnit, directory, fatal, retries = 0, Closure body) {
```

Reply:
Good suggestion. Addressed with commit https://github.com/xamarin/xamarin-android/pull/2797/commits/f8b16d4c39c45ed39e853e70de61a1f4dbf85fdf
I kept the variable name as ctAttempts since the parameter to the retry statement represents the total number of attempts that will be made; not the number of retries after the initial attempt. A value of 0 or 1 leads to one initial attempt.